### PR TITLE
SNOW-859635: Retry Strategy v2

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-jy**
### Description
- [x] Edited default values of the retry options.
- [x] Replaced the option name 'loginTimeout' with 'retryTimeout'

### Checklist
- [ ] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [ ] Create tests which fail without the change (if possible)
- [ ] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [ ] Extend the README / documentation and ensure is properly displayed (if necessary)
- [ ] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: lib/connection/connection_config.js [hunk 1]
```diff
@@ -485,12 +485,12 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo)
...
-  if (Util.exists(options.loginTimeout)) {
-    Errors.checkArgumentValid(Util.isNumber(options.loginTimeout),
-    ErrorCodes.ERR_CONN_CREATE_INVALID_MAX_LOGIN_TIMEOUT);
+  let retryTimeout = 300;
+  if (Util.exists(options.retryTimeout)) {
+    Errors.checkArgumentValid(Util.isNumber(options.retryTimeout),
+    ErrorCodes.ERR_CONN_CREATE_INVALID_MAX_RETRY_TIMEOUT);
 
-    loginTimeout = Math.max(loginTimeout,options.loginTimeout);
+    retryTimeout = Math.max(retryTimeout,options.retryTimeout);
```
Comment (sfc-gh-ext-simba-hx) [inline lib/connection/connection_config.js:493]: Please allow it to be set to 0 as unlimited.

File: lib/util.js [hunk 2]
```diff
@@ -431,8 +431,8 @@ exports.nextSleepTime = function (
  * @returns {JSON} return next sleep Time and totalTime.
  */
 exports.jitteredSleepTime = function (numofRetries, currentSleepTime, totalTimeout, maxLoginTimeout) {
```
Comment (sfc-gh-ext-simba-hx) [inline lib/util.js:433]: better to change the argument name to maxRetryTimeout to avoid confusing.

File: lib/util.js [hunk 3]
```diff
@@ -444,8 +444,8 @@ exports.jitteredSleepTime = function (numofRetries, currentSleepTime, totalTimeo
  * @returns {Boolean} return a random number between two numbers.
  */
 function chooseRandom (firstNumber, secondNumber) {
-  let bigNumber = Math.max(firstNumber, secondNumber);
-  let smallNumber = Math.min(firstNumber, secondNumber);
+  const bigNumber = Math.max(firstNumber, secondNumber);
+  const smallNumber = Math.min(firstNumber, secondNumber);
```
Comment (sfc-gh-ext-simba-hx) [inline lib/util.js:448]: It doesn't really matter which one is bigger. 
Math.random() * (firstNumber - secondNumber)) + secondNumber
would always return what we need.

Comment (sfc-gh-ext-simba-jy) [reply lib/util.js]: Fixed

File: lib/connection/connection_config.js (hunk 1 above)
Comment (sfc-gh-ext-simba-jy) [reply lib/connection/connection_config.js]: fixed

File: lib/util.js (hunk 2 above)
Comment (sfc-gh-ext-simba-jy) [reply lib/util.js]: Replaced.

**Review by sfc-gh-ext-simba-hx — APPROVED** [review]
LGTM. Please ask Snowflake for review.

**Review by sfc-gh-pmotacki — APPROVED** [review]
Please reformat code in some places.
Rest LGTM

**Comment by sfc-gh-pmotacki** [comment]
LGTM

**Pull request merged**